cs/src/Management/TunnelManagementClient.cs

```diff
@@ -140,14 +163,23 @@ public class TunnelManagementClient : ITunnelManagementClient
         /// <see cref="HttpClientHandler"/> specified (or at the end of the chain) must have
         /// automatic redirection disabled. The provided HTTP handler will not be disposed
         /// by <see cref="Dispose"/>.</param>
+        /// <param name="apiVersion"> Api version to use for tunnels requests, accepted
+        /// values are <see cref="TunnelsApiVersions"/></param>
         public TunnelManagementClient(
             ProductInfoHeaderValue[] userAgents,
             Func<Task<AuthenticationHeaderValue?>>? userTokenCallback = null,
             Uri? tunnelServiceUri = null,
-            HttpMessageHandler? httpHandler = null)
+            HttpMessageHandler? httpHandler = null,
+            string? apiVersion = null)
         {
             Requires.NotNullEmptyOrNullElements(userAgents, nameof(userAgents));
             UserAgents = Requires.NotNull(userAgents, nameof(userAgents));
+            if (!TunnelsApiVersions.Contains(apiVersion))
```

Comment:
By default `apiVersion` is null. However, `TunnelsApiVersions` doesn't contain `null` and the default value will always fail. Did you mean to allow `null` as a valid value, or always require non-null value? If the latter, the ctor argument must not allow nulls.

Reply:
Added null as a vaild in the check